subcommands/devices/list.go

```diff
@@ -220,24 +220,26 @@ func doList(cmd *cobra.Command, args []string) {
 	sortBy = appendSortFlagValue(sortBy, cmd, "sort-by-last-seen", "last_seen")
 	sortBy = appendSortFlagValue(sortBy, cmd, "sort-by-name", "name")
 
-	name_ilike := ""
+	filterBy := map[string]string{
+		"factory":     factory,
+		"group":       deviceByGroup,
+		"match_tag":   deviceByTag,
+		"target_name": deviceByTarget,
+		"uuid":        deviceUuid,
+	}
 	if len(args) == 1 {
-		name_ilike = sqlLikeIfy(args[0])
-	}
-	dl, err := api.DeviceList(
-		deviceMine,
-		deviceOnlyProd,
-		deviceOnlyNonProd,
-		deviceByTag,
-		factory,
-		deviceByGroup,
-		name_ilike,
-		deviceUuid,
-		deviceByTarget,
-		strings.Join(sortBy, ","),
-		showPage,
-		paginationLimit,
-	)
+		filterBy["name_ilike"] = sqlLikeIfy(args[0])
+	}
+	if deviceMine {
+		filterBy["mine"] = "1"
+	}
+	if deviceOnlyProd {
+		filterBy["prod"] = "1"
+	} else if deviceOnlyNonProd {
+		filterBy["prod"] = "0"
+	}
+
+	dl, err := api.DeviceList(filterBy, strings.Join(sortBy, ","), showPage, paginationLimit)
```

Comment:
I'll give it an LGTM. I wonder if using a DeviceOpts struct wouldn't be a little cleaner assuming there's a relatively easy way to convert the struct into a dict net/url.Values ?

Reply:
Using `DeviceOpts` was my initial idea, but I decided that a simple `map[string]string` would result into less code churn long term.

As for the easiness converting struct to map, I'm not sure... One "easy" way I know is to do `json.Marshal/json.Unmarshal`, maybe there is also something inside the `reflect` package. But those are not easy IMHO. No wonder there are [3PP packages](https://github.com/fatih/structs) to do just that. Could you elaborate what do you meant by "easy" here.